data-prepper-plugins/mutate-event-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutateevent/CopyValueProcessor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.mutateevent;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.processor.AbstractProcessor;
+import com.amazon.dataprepper.model.processor.Processor;
+import com.amazon.dataprepper.model.record.Record;
+
+import java.util.Collection;
+
+@DataPrepperPlugin(name = "copy_value", pluginType = Processor.class, pluginConfigurationType = CopyValueProcessorConfig.class)
+public class CopyValueProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final String key;
+    private final String newKey;
+    private final boolean overwriteIfToKeyExists;
+
+    @DataPrepperPluginConstructor
+    public CopyValueProcessor(final PluginMetrics pluginMetrics, final CopyValueProcessorConfig config) {
+        super(pluginMetrics);
+        this.key = config.getFromKey();
+        this.newKey = config.getToKey();
+        this.overwriteIfToKeyExists = config.getOverwriteIfToKeyExists();
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        if (key.equals(newKey)) {
```

Comment:
Considering it's very likely that this would be a typo on the user's part, I think it makes sense to validate and give a nice configuration error message.

Reply:
I had considered that myself. The only problem I see is that the exception message wouldn't be able to pinpoint the exact location of the error if there were a large list of operations. It'd only be able to say something like, "to_key and from_key cannot be the same value"

I can still add in a quick Jakarta validation if desired.